assets/app/view/game/buy_trains.rb

```diff
@@ -229,12 +230,17 @@ def from_depot(depot_trains)
                 warranties: warranties,
               ))
             end
-
+            train_props = { style: {} }
+            unless @game.licensed_to_operate?(corporation, train, name)
```

Comment:
I would prefer a more generic name for this method that is not tied to the 1862 concept of licenses.

Reply:
I've changed it to able_to_operate which should be generic enough